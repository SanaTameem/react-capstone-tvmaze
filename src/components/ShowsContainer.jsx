import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import '../styles/ShowsContainer.css';
import { useSelector, useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCircleArrowRight } from '@fortawesome/free-solid-svg-icons';
import { fetchShow } from '../redux/Show/ShowSlice';

function ShowsContainer() {
  const dispatch = useDispatch();
  const showsData = useSelector((state) => state.show.shows);
  useEffect(() => {
    if (showsData.length === 0) {
      dispatch(fetchShow());
    }
  }, [dispatch]);

  return (
    <div className="main-container">
      <div className="search-bar">
        <form className="form">
          <input
            type="search"
            className="search-box"
            placeholder="Search by show name"
          />
        </form>
      </div>

      <div className="cards-container">
        {showsData.map((item) => (
          <div className="show-card" key={item.id}>
            <div className="card-img">
              <img src={item.image} alt="show-imgs" />
              <Link to={`/showDetails/${item.id}`}>
                <FontAwesomeIcon icon={faCircleArrowRight} className="next-icon" />
              </Link>
            </div>
            <div className="home-details">
              <p className="name">
                Name :
                {' '}
                {item.name}
              </p>
              <p className="rating">
                Rating :
                {' '}
                {item.rating}
              </p>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}

export default ShowsContainer;
