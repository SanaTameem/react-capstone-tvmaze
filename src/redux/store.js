import { configureStore } from "@reduxjs/toolkit";
import showsReducer from './Show/ShowSlice'
const store = configureStore({
    reducer:{
        show: showsReducer,
    }
});

export default store;