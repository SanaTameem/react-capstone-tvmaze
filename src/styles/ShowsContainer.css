.main-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}

.search-bar {
  width: 100%;
  display: flex;
  justify-content: center;
  padding: 10px;
  background-color: rgb(97, 148, 231);
  padding-bottom: 20px;
}

.form {
  width: 30%;
}

.search-box {
  width: 100%;
  padding: 10px;
  border-radius: 22px;
  border: none;
  font-size: 20px;
  color: rgb(97, 148, 231);
}

.search-box::placeholder {
  text-align: center;
  font-size: 18px;
  color: #9c9b9b;
}

.cards-container {
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  border: 2px solid #000;
}

.show-card {
  padding: 5px;
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 100%;
  border: 2px solid #000;
}

.card-img {
  display: flex;
  justify-content: flex-end;
  width: 80%;
  height: 240px;
  background-repeat: no-repeat;
  background-size: 100% 240px;
}

.card-img img {
  width: 100%;
  height: 100%;
}

.next-icon {
  font-size: 30px;
  color: #fff;
  padding: 8px;
}

.home-details {
  display: flex;
  flex-direction: column;
  width: 100%;
  padding: 10px;
  color: #fff;
  align-items: flex-end;
  font-size: 15px;
  margin-right: 60px;
}

.name {
  width: 100%;
  font-size: 20px;
  font-weight: 700;
  text-align: right;
}

@media screen and (max-width: 768px) {
  .cards-container {
    display: grid;
    grid-template-columns: 1fr 1fr;
  }

  .card-img {
    background-size: 100% 135px;
    width: 85%;
    height: 200px;
    height: 135px;
  }

  .card-img img {
    width: 94%;
    height: 95%;
  }

  .next-icon {
    font-size: 18px;
  }

  .logo {
    width: 30%;
  }

  .nav-title {
    width: 62%;
    font-size: 24px;
  }

  .navbar {
    justify-content: space-between;
    padding: 0 13px;
  }

  .setting-mic {
    width: 20%;
  }

  .form {
    width: 90%;
  }

  .name {
    font-size: 16px;
  }

  .home-details {
    font-size: 12px;
    margin-right: 12px;
  }
}

@media screen and (min-width: 768px) and (max-width: 1024px) {
  .cards-container {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
  }

  .card-img {
    width: 76%;
    height: 221px;
  }

  .next-icon {
    font-size: 25px;
  }

  .form {
    width: 50%;
  }

  .nav-title {
    width: 50%;
  }
}
