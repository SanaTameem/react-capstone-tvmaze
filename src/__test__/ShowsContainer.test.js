import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import ShowsContainer from '../components/ShowsContainer';
import store from '../redux/store';
import '@testing-library/jest-dom';

describe('ShowsContainer component test', () => {
  const Bitten = {
    name: 'Bitten',
    rating: 7.5,
    id: 3,
  };

  it('Check if the component changed', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <ShowsContainer
              key={Bitten.id}
              id={Bitten.id}
              name={Bitten.name}
            />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
