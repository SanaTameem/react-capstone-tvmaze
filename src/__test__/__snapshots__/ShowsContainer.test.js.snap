// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowsContainer component test Check if the component changed 1`] = `
<div
  className="main-container"
>
  <div
    className="search-bar"
  >
    <form
      className="form"
    >
      <input
        className="search-box"
        onChange={[Function]}
        placeholder="Search by show name e.g(the 100)"
        type="search"
        value=""
      />
    </form>
  </div>
  <div
    className="cards-container"
  />
</div>
`;
